realm-jni/src/io_realm_internal_UncheckedRow.cpp

```diff
@@ -294,17 +302,29 @@ JNIEXPORT void JNICALL Java_io_realm_internal_UncheckedRow_nativeSetByteArray
   (JNIEnv* env, jobject, jlong nativeRowPtr, jlong columnIndex, jbyteArray value)
 {
     TR_ENTER_PTR(nativeRowPtr)
+
     if (!ROW_VALID(env, ROW(nativeRowPtr)))
         return;
 
-    jbyte* bytePtr = env->GetByteArrayElements(value, NULL);
-    if (!bytePtr) {
-        ThrowException(env, IllegalArgument, "doByteArray");
-        return;
-    }
-    size_t dataLen = S(env->GetArrayLength(value));
-    ROW(nativeRowPtr)->set_binary( S(columnIndex), BinaryData(reinterpret_cast<char*>(bytePtr), dataLen));
-    env->ReleaseByteArrayElements(value, bytePtr, 0);
+    try {
+        if (value == NULL) {
+            if (!(ROW(nativeRowPtr)->get_table()->is_nullable(S(columnIndex)))) {
+                ThrowNullValueException(env, ROW(nativeRowPtr)->get_table(), S(columnIndex));
+                return;
+            }
+            ROW(nativeRowPtr)->set_binary(S(columnIndex), BinaryData());
+        }
+        else {
+            jbyte* bytePtr = env->GetByteArrayElements(value, NULL);
+            if (!bytePtr) {
+                ThrowException(env, IllegalArgument, "doByteArray");
+                return;
+            }
+            size_t dataLen = S(env->GetArrayLength(value));
+            ROW(nativeRowPtr)->set_binary( S(columnIndex), BinaryData(reinterpret_cast<char*>(bytePtr), dataLen));
+            env->ReleaseByteArrayElements(value, bytePtr, 0);
+        }
+    } CATCH_STD()
```

Comment:
We are not that consistent: `io_realm_internal_table.cpp` has `CATCH_STD()` on the same line.